src/models/IDPDiscovery.js

```diff
@@ -54,9 +64,10 @@ export default PrimaryAuthModel.extend({
       .webfinger(webfingerArgs)
       .then(res => {
         if (res && res.links && res.links[0]) {
-          if (res.links[0].properties['okta:idp:type'] === 'OKTA') {
+          if (res.links[0].properties['okta:idp:type'] === 'OKTA' && !autoTriggered) {
             this.trigger('goToPrimaryAuth');
-          } else if (res.links[0].href) {
+          } else if (res.links[0].properties['okta:idp:type'] !== 'OKTA' && res.links[0].href) {
+            this.showSpinner(autoTriggered);
```

Comment:
Without this code, we'll show the username field until we redirect the browser to the idp. I do see the username field a fraction of a second before we show the spinner and then eventually redirect to the idp with this code as it is now. @haishengwu-okta do you have a better suggestion for how to avoid showing the username field if we are redirecting?